projects/canopy/src/lib/modal/modal.component.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  AfterContentInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ContentChild,
+  EventEmitter,
+  forwardRef,
+  HostListener,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output,
+} from '@angular/core';
+
+import { Subscription } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { keyName } from '../utils/keyboard-keys';
+
+import { LgModalService } from './modal.service';
+import { LgModalHeaderComponent } from './modal-header/modal-header.component';
+import { LgModalBodyComponent } from './modal-body/modal-body.component';
+
+@Component({
+  selector: 'lg-modal',
+  templateUrl: './modal.component.html',
+  styleUrls: ['./modal.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class LgModalComponent implements OnInit, AfterContentInit, OnDestroy {
+  @Input() id: string;
+  @Input() closeOnOverlayClick = true;
+  @Output() open: EventEmitter<void> = new EventEmitter();
+  @Output() closed: EventEmitter<void> = new EventEmitter();
+  @ContentChild(forwardRef(() => LgModalHeaderComponent))
+  modalHeaderComponent: LgModalHeaderComponent;
+  @ContentChild(forwardRef(() => LgModalBodyComponent))
+  modalBodyComponent: LgModalBodyComponent;
+
+  isOpen: boolean;
+  subscriptions: Array<Subscription> = [];
+
+  constructor(private cdr: ChangeDetectorRef, private modalService: LgModalService) {}
+
+  ngOnInit(): void {
+    this.subscriptions.push(
+      this.modalService
+        .isOpen$(this.id)
+        .pipe(
+          map((isOpen) => {
+            this.isOpen = isOpen;
+
+            const bodyEl: HTMLBodyElement = document.querySelector('body');
+
+            if (isOpen) {
+              bodyEl.style.overflow = 'hidden';
+              this.open.emit();
+            } else {
+              this.closed.emit();
+              bodyEl.style.overflow = '';
+            }
+
+            this.cdr.detectChanges();
+          }),
+        )
+        .subscribe(),
+    );
+  }
+
+  ngAfterContentInit(): void {
+    this.modalHeaderComponent.id = `lg-modal-header-${this.id}`;
+    this.modalHeaderComponent.modalId = this.id;
+    this.modalBodyComponent.id = `lg-modal-body-${this.id}`;
+  }
+
+  @HostListener('keydown', ['$event']) onKeydown(event: KeyboardEvent): void {
+    if (event.key === keyName.KEY_ESCAPE && this.isOpen && this.closeOnOverlayClick) {
```

Comment:
Just wondering if `this.closeOnOverlayClick` needs to be `true` for this one? Is it an accessibility issue to not allow a user to dismiss it with the ESC key? Or is this for instances where the user has to click on a specific button, or perform a specific action to get past the modal?

Reply:
Sorry I'm not sure I completely understand.
It's a requirement that if the overlay is dismissible by clicking outside the modal a keyboard user should have the same functionality by clicking `ESC`. Same is for the opposite: if a specific action is required on the modal and we don't want the user to dismiss the modal with a click outside then the same should happen for the keyboard user.

Does any of the above clarifies your questions?

Reply:
Yes thanks, I also saw another comment below about the user making a decision on some modals, that makes sense as well 👍